### PR TITLE
Feature/particle

### PR DESCRIPTION
**Pull request by whaeck**
Updated isotope identifier to be case insensitive, made mass optional for elemental isotope.

Added the string based particle identifier constructor (cascading try ... catch ...)

### CODE REVIEW

File: src/elementary/IsotopeID/test/IsotopeID.test.cpp [hunk 1]
```diff
@@ -51,13 +57,31 @@ SCENARIO( "IsotopeID" ) {
...
+      id = IsotopeID( "h0" );
+      CHECK( ElementID( 1 ) == id.element() );
+      CHECK( 0 == id.mass() );
+      CHECK( "H0" == id.symbol() );
+      CHECK( "H0" == id.name() );
+
+      id = IsotopeID( "H" );
+      CHECK( ElementID( 1 ) == id.element() );
+      CHECK( 0 == id.mass() );
+      CHECK( "H0" == id.symbol() );
```
Comment (jlconlin) [inline src/elementary/IsotopeID/test/IsotopeID.test.cpp:70]: I thought that `H` meant hydrogen (the element) and `h` meant hydrogen (an isotope). It seems with this Pull Request, this definition has changed somewhat. Am I mistaken?

File: src/elementary/ParticleID.hpp [hunk 2]
```diff
@@ -22,12 +22,15 @@ namespace elementary {
   class ParticleID {
 
     /* type aliases */
-    using IdentifierVariant = std::variant< FundamentalParticleID, NucleusID, NuclideID >;
+    using IdentifierVariant = std::variant< FundamentalParticleID,
```
Comment (jlconlin) [inline src/elementary/ParticleID.hpp:25]: :smile:

File: src/elementary/ParticleID/test/ParticleID.test.cpp [hunk 3]
```diff
@@ -24,11 +24,23 @@ SCENARIO( "ParticleID" ) {
...
+      id = ParticleID( "h2" );
+      CHECK( "h2" == id.symbol() );
+      CHECK( "h2" == id.name() );
+
+      id = ParticleID( NuclideID( "H2" ) );
+      CHECK( "H2" == id.symbol() );
+      CHECK( "H2" == id.name() );
+
+      id = ParticleID( "H2" );
       CHECK( "H2" == id.symbol() );
```
Comment (jlconlin) [inline src/elementary/ParticleID/test/ParticleID.test.cpp:44]: At some point we need to test for the failure case.

File: src/elementary/IsotopeID/test/IsotopeID.test.cpp (hunk 1 above)
Comment (whaeck) [reply src/elementary/IsotopeID/test/IsotopeID.test.cpp:70]: Yes, I changed that following our discussion from a few days ago. As long as the final symbol has the 0, it should be fine. It was in the pull request description though ;-)

File: src/elementary/ParticleID/test/ParticleID.test.cpp (hunk 3 above)
Comment (whaeck) [reply src/elementary/ParticleID/test/ParticleID.test.cpp:44]: I'll have to add a few.

**Pull request merged**